matgen/generate_type_gfpp.hh

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2017-2022, University of Tennessee. All rights reserved.
+// SPDX-License-Identifier: BSD-3-Clause
+// This program is free software: you can redistribute it and/or modify it under
+// the terms of the BSD 3-Clause license. See the accompanying LICENSE file.
+
+#ifndef SLATE_GENERATE_TYPE_GFPP_HH
+#define SLATE_GENERATE_TYPE_GFPP_HH
+
+
+#include "slate/slate.hh"
+#include "test.hh"
+#include "print_matrix.hh"
+
+#include <exception>
+#include <string>
+#include <vector>
+#include <limits>
+#include <complex>
+#include <chrono>
+
+#include <cmath>
+#include <cstdio>
+#include <cstdlib>
+#include <utility>
+
+#include "matrix_params.hh"
+#include "generate_matrix.hh"
+#include "random.hh"
+
+
+namespace slate {
+
+
+void generate_gfpp( slate::Matrix<scalar_t>& A,
```

Comment:
We can implement `gfpp` with `set` and the following lambda.
```C++
n_1 = A.n() - 1;
[n] (int64_t i, int64_t j) {
    if (j == n_1) { // last column
          return 1.0;
    }
    else if (i > j) { // below the diagonal
        return -1.0;
    }
    else if (i == j) { // diagonal
        return 0.5;
    }
    else { // above the diagonal
        return 0.0;
    }
}
```

Reply:
Implemented. Thank you for the comment.